src/app/GitCommands/UserRepositoryHistory/RepositoryDescriptionProvider.cs

```diff
@@ -65,6 +77,34 @@ public string Get(string repositoryDir)
             return dirInfo.Name;
         }
 
+        public string GetDescriptiveUnique(string repositoryDir)
+        {
+            string unique = GetUnique(repositoryDir);
+
+            ReadOnlySpan<char> path = repositoryDir.AsSpan();
```

Comment:
If .git/description exists, it must be used, set in line 91
Maybe no duplicate check even, use the name the user wants?

```suggestion
            string descName = Get(repositoryDir;
            ReadOnlySpan<char> path = repositoryDir.AsSpan();
```